scripts/scil_mrds_select_number_of_tensors.py

```diff
@@ -0,0 +1,172 @@
+    g.add_argument('--N1', nargs=5, required=True,
+                   help='MRDS solution with 1 tensor.')
+    g.add_argument('--N2', nargs=5, required=True,
+                   help='MRDS solution with 2 tensors.')
+    g.add_argument('--N3', nargs=5, required=True,
+                   help='MRDS solution with 3 tensors.')
+
+    p.add_argument('--prefix', default='results',
+                   help='prefix of the MRDS results [%(default)s].')
+    p.add_argument('--mask',
+                   help='Optional mask filename.')
+
+    add_processes_arg(p)
+    add_sh_basis_args(p)
+    add_verbose_arg(p)
+    add_overwrite_arg(p)
+    return p
+
+
+def main():
+    parser = _build_arg_parser()
+    args = parser.parse_args()
+    logging.getLogger().setLevel(args.verbose.upper())
+
+    assert_inputs_exist(parser, [args.in_volume] + args.N1 + args.N2 + args.N3,
+                        optional=args.mask)
+    output_files = ["{}_MRDS_signal_fraction.nii.gz".format(args.prefix),
+                    "{}_MRDS_eigenvalues.nii.gz".format(args.prefix),
+                    "{}_MRDS_isotropic.nii.gz".format(args.prefix),
+                    "{}_MRDS_num_comp.nii.gz".format(args.prefix),
+                    "{}_MRDS_evecs.nii.gz".format(args.prefix)]
+    assert_outputs_exist(parser, args, output_files)
+    assert_headers_compatible(parser, [args.in_volume] +
+                              args.N1 + args.N2 + args.N3)
+
+    mrds_files = [args.N1, args.N2, args.N3]
+
+    signal_fraction = []
+    eigenvalues = []
+    iso = []
+    num_tensors = []
+    evecs = []
+    for N in range(3):
+        signal_fraction.append(nib.load(mrds_files[N][0]).get_fdata(dtype=np.float32))
+        eigenvalues.append(nib.load(mrds_files[N][1]).get_fdata(dtype=np.float32))
+        iso.append(nib.load(mrds_files[N][2]).get_fdata(dtype=np.float32))
+        num_tensors.append(nib.load(mrds_files[N][3]).get_fdata(dtype=np.float32))
+        evecs.append(nib.load(mrds_files[N][4]).get_fdata(dtype=np.float32))
+
+    # MOdel SElector MAP
+    mosemap_img = nib.load(args.in_volume)
+    mosemap = get_data_as_labels(mosemap_img)
+    header = mosemap_img.header
+
+    affine = mosemap_img.affine
+    X, Y, Z = mosemap.shape[0:3]
+
+    # load mask
+    if args.mask:
+        mask = get_data_as_mask(nib.load(args.mask), dtype=bool)
+    else:
+        mask = np.ones((X, Y, Z), dtype=np.uint8)
+
+    # select data using mosemap
+    voxels = itertools.product(range(X), range(Y), range(Z))
+    filtered_voxels = ((x, y, z) for (x, y, z) in voxels if mask[x, y, z])
+
+    signal_fraction_out = np.zeros((X, Y, Z, 3))
+    eigenvalues_out = np.zeros((X, Y, Z, 9))
+    iso_out = np.zeros((X, Y, Z, 2))
+    num_tensors_out = np.zeros((X, Y, Z), dtype=np.uint8)
+    evecs_out = np.zeros((X, Y, Z, 9))
+
+    # select data using mosemap
+    for (X, Y, Z) in filtered_voxels:
+        N = mosemap[X, Y, Z]-1
+
+        if N > 2:
+            N = 2
```

Comment:
Should this be explained somewhere? Nufo values will be clipped? Or is it part of the MRDS theory?